### ISSUE TITLE
Is it necessary to support publish=yes apart from saying access != 'dark' ?

### ISSUE THREAD

**Issue by jcoyne** [sul-dlss/dor-services-app#2407]
Currently cocina simplifies access controls by making the assumption that if a file is dark, then publish="no" and if publish="no" then the file is dark.  This works for most objects however there may be some corner cases that this does not handle.

It's possible that WARCs, ETDs or 3d set publish="no" without intending for the files to be dark.

We show the viewer if any file is published even if the viewewer doesn't support that file type, so one work around is to mark the file as not published.

Currently, on the ToFedora side of the Cocina mappings, we derive the contentMetadata publish attribute from the access settings for a particular file. If the file is not dark, publish is set to yes, else no.

**Comment by jcoyne** [comment]
Note: this is the access level on the file itself, not the access set on the object.

**Comment by andrewjbtw** [comment]
Examples:

Recent ETDs: 

https://argo.stanford.edu/view/druid:ww428zp8348 (original submitted PDF is shelve/preserve)
https://argo.stanford.edu/view/druid:kj020rh6303 (original submitted PDF + permissions file is shelve/preserve)

Recent web archive crawls:

https://argo.stanford.edu/view/druid:bp127tc5809 (object rights are world and file is shelve/preserve)
https://argo.stanford.edu/view/druid:hx172pg3079 (object rights are dark and files are shelve/preserve)

3d:

https://argo.stanford.edu/view/druid:bc769sr4504 (some files are only shelve/preserve)

**Comment by jcoyne** [comment]
@andrewjbtw when I say "access = dark" I'm not talking about object rights, I'm talking about file rights. (cocina has both)

**Comment by andrewjbtw** [comment]
I'm aware of that. The Fedora model also has file-level rights. Those rights can be any of the rights settings supported at the object level (world, stanford, no-download, location-restricted, dark). In the cases above, the file-level rights are the same as the object-level rights, but are not called out explicitly in the rightsMetadata. Only file-level rights that differ from object-level rights are called out explicitly in the Fedora model.

Where I understand the difference, Cocina makes assumptions about the relationship between the publish setting and the file-level rights setting. The Fedora model does not.

So if "publish" = "no" for a file in the Fedora model the file-level access can still be "world" downloadable (or any of the other rights settings).

But if "publish" = "no" for a file in the Cocina model it sounds like that makes the file-level access "dark"? That could be a problem for systems that need to read the file even though it is not published for human users to see.

On the flip side, if the assumption is made that a file-level access of "!dark" means "publish" = "yes", then the system will present files for download in the viewer that users might not have intended to list. The "publish" setting isn't solely about whether a file appears in the Purl XML, it also affects what files are in the download list presented by the viewer. This list of files is part of the user experience.

In the ETD example above, an assumption that "publish"="yes" for the shelved, non-dark files would present all those files for download. I don't think we'd want to make the submitted dissertation or the permissions files available like that. But I also don't know why they're shelved at all.

In the web archive example above, an assumption that "publish"="yes" for the shelved, non-dark files would list the WARCs for download. I don't think that's the outcome we want. But I also don't know why those files are shelved. It might have something to do with how SWAP reaches the files.

There are other use cases where people either set publish in contentMetadata and/or set file-level rights in the rightsMetadata in custom ways, or where they would like to do so but give up on it because the UI is too difficult to deal with. Taking away the ability to set publish at all would not be an improvement.

The tl;dr is that users have a need to be able to set all of "publish", "shelve",  "preserve", and file-level rights on a file by file basis. At least as long as the access-side systems expect the current model.
________________________________

**Comment by mjgiarlo** [comment]
I assume that once we merge https://github.com/sul-dlss/cocina-models/pull/222, we can close this issue, so I've linked them. LMK if that's not right, @jcoyne @andrewjbtw

**Comment by jcoyne** [comment]
I don't think we've actually done a good job of answering this yet... but I'm not sure we're going to either.

I think the complexity of interplay between access and publish makes this system much harder for our users than it has to be, however, we've come to depend on this complexity. 🤷
